test/compiles_test.dart

```diff
@@ -33,6 +33,22 @@ void main() {
     expectTestPassed(liveTest);
   });
 
+  test('supports relative path dependencies', () async {
```

Comment:
I don't understand what this test is trying to do. Could you explain to me?

Reply:
Sure! The test is asserting that resolution of relative path dependencies works, e.g.:

```yaml
dependencies:
  expect_error:
    path: ..
```

When `pub` resolves such a dependency, it writes a relative path in to `package_config.json`. The path is not relative to the root package directory, though, but is instead relative to `package_config.json`.  

I've added such a dependency to `flutter_package` and the test declares an import on `expect_error` to check if the analyzer can correctly resolve the imported package. 

If you revert the changes in `Library.custom` the test fails with the following error:
```
No expect-error found for code URI_DOES_NOT_EXIST but an error was found: Target of URI doesn't exist: 'package:expect_error/expect_error.dart'.
```

Reply:
@rrousselGit Did that clear it up?